### PR TITLE
Implemented GET endpoint for GlobalPath Interface

### LINKED ISSUE #21: [API] Implement GET endpoint for GlobalPath Interface
### Purpose

Endpoint to retrieve GlobalPath data from the database. 

### Description

HTTP request URL: `localhost:3005/api/globalpath`  

Some guidelines:
- Next.js automatically create routes by following the directory structure under `pages`. For example, `pages/api/gps` will create the route `localhost:3005/api/gps`. 
- The implementation will be similar to what was done here: https://github.com/UBCSailbot/website/blob/main/pages/api/gps/index.ts
- Make sure to write tests. You can take a look at https://github.com/UBCSailbot/website/pull/19 for inspiration. (Particularly look at the files under the directories `tests/steps` and `tests/features` 
- For the tests, please make sure the global waypoints you insert into DB is realistic and matches what we have here: https://docs.google.com/presentation/d/1ewzopIbE7nNtew-pzOVGlbBm-ShRoF-1CaN7ZTIj9UU/edit#slide=id.g24c9a897be5_0_181 (Slide 4) 
- Rely on https://www.google.com/maps, just click on the water and it should give you the lat/lon points. 

### Resources

- https://nextjs.org/docs/pages/building-your-application/routing/pages-and-layouts

### PR DESCRIPTION
**Pull request by dk1702**
### Description

- Resolves #21 
- Implementation was done very similarly to #15 & #22

### Verification

- [ ] 

### Resources

- #15 
- #22

### CODE REVIEW

File: tests/steps/given.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { Given, Then } from "@cucumber/cucumber";
 import GPS from '@/models/GPS';
 import ConnectMongoDB from '@/lib/mongodb';
 import AISShips from '@/models/AISShips';
+import GlobalPath from '@/models/GlobalPath';
 
 Given('I clear the database', async function () {
     const db = await ConnectMongoDB();
```
Comment (jahn18) [inline tests/steps/given.ts:10]: Please add a line below to clear all the globalpath data (following what GPS and AISShips do)

File: tests/steps/given.ts [hunk 2]
```diff
@@ -50,6 +51,29 @@ Given('I insert AISShips data into the database', async function () {
     await AISShips.create(aisshipData);
 });
 
+Given('I insert GlobalPath data into the database', async function () {
+    const globalpathData = {
```
Comment (jahn18) [inline tests/steps/given.ts:55]: Camel casing please

**Review by jahn18 — COMMENTED** [review]
Good work!

File: tests/steps/given.ts (hunk 1 above)
Comment (dk1702) [reply tests/steps/given.ts:10]: I added `await GlobalPath.deleteMany();` underneath, but some GlobalPath data is still showing up when going to the endpoint - am I missing something else ? picture of data is attached below: 

image

Comment (jahn18) [reply tests/steps/given.ts:10]: Can you run it a couple of times consecutively? If it only returns one document with the same three waypoints, then it's working. 

If you didn't have the `await GlobalPath.deleteMany()`, then if you were to run the test multiple times, the documents would stack up.

File: models/GlobalPath.ts [hunk 3]
```diff
@@ -11,9 +15,23 @@ export interface GlobalPath extends mongoose.Document {
 
 const GlobalPathSchema = new mongoose.Schema<GlobalPath>({
     waypoints: {
-        type: WayPoint[],
+        type: [
+            {
+                id:Number,
```
Comment (jahn18) [inline models/GlobalPath.ts:20]: Please remove `id` here. It doesn't exist in the GlobalPath interface

**Review by jahn18 — APPROVED** [review]
Looks good to me

**Pull request merged**